### PR TITLE
Exporting some Dynamic Mode stats

### PR DESCRIPTION
**Pull request by DeityLink**
And fixing a dumb bug

### CODE REVIEW

File: code/datums/statistics/stat_helpers.dm [hunk 1]
```diff
@@ -162,13 +162,13 @@
 	// 	var/datum/game_mode/mixed/mixy = ticker.mode
 	// 	for(var/datum/game_mode/GM in mixy.modes)
 	// 		mixed_gamemodes.Add(GM.name)
-
+	//
 	for(var/datum/mind/M in ticker.minds)
 		add_objectives(M)
 		if(istype(M.current, /mob/living) && !M.current.isDead())
 			add_survivor_stat(M.current)
-			if(M.special_role == "Cultist")
```
Comment (gbasood) [inline code/datums/statistics/stat_helpers.dm:170]: This won't ever apply again so you can just remove the commented-out cultist check

File: code/datums/statistics/statcollection.dm [hunk 2]
```diff
@@ -218,6 +223,7 @@ proc/datum2json(var/datum/D, var/list/do_not_copy=datum_donotcopy)
 
 	var/jsonout = datum2json(src)
 	statfile << jsonout
+	//to_chat(world,jsonout)
```
Comment (gbasood) [inline code/datums/statistics/statcollection.dm]: remove this commented code

File: code/datums/gamemode/dynamic/dynamic.dm [hunk 3]
```diff
@@ -220,15 +245,20 @@ var/list/forced_roundstart_ruleset = list()
...
-			for (var/datum/dynamic_ruleset/latejoin/rule in midround_rules)
+			for (var/datum/dynamic_ruleset/midround/rule in midround_rules)
 				if (rule.acceptable(living_players.len,threat_level) && threat >= rule.cost)
 					rule.candidates = current_players.Copy()
 					rule.trim_candidates()
 					if (rule.ready())
 						drafted_rules[rule] = rule.weight
 
 			if (drafted_rules.len > 0)
+				message_admins("DYNAMIC MODE: [drafted_rules.len] elligible rulesets.")
```
Comment (gbasood) [inline code/datums/gamemode/dynamic/dynamic.dm:256]: \>\>\>\>UNATOMIC
don't actually care just jokes

**Review by gbasood — CHANGES_REQUESTED** [review]
Overall looks sound, although the stat code is scattered throughout a lot
Remove the commented code I pointed out

Important questions:
 - Did you test this?
 - Do you have an example of the JSON file that would be exported at round end? Because I, or whoever else makes the stats server process the resulting file, would want that as reference for when they write the processor code on the web server side.

**Comment by DeityLink** [comment]
@gbasood 
image
1. yep the above pic was the resulting json code pasted into http://jsonviewer.stack.hu/
2. I'll send you on Discord

**Comment by DeityLink** [comment]
So basically the idea is to have all 5 population levels stacked on the same graph, and a vertical red line or something to mark successful_injections

**Comment by gbasood** [comment]
Okay so:
1. For `pop_levels` objects, the format should be more like:
 - `dynamic_stats` top object, like it already is
 - array of objects instead of objects directly on the `pop_levels` object
 - The pop_level objects should have named variables rather than being an array.

Same for threat levels and successful injections. Array of objects with named variables.

Example:
```json
{
  "dynamic_stats": {
    "threat_levels": [
      {
        "time": "2018-09-29 22:17:42",
        "threat": 67.6
      }
    ],
    "pop_levels": [
      {
        "time": "2018-09-29 22:17:42",
        "total_server_pop": 1,
        "living_players": 1,
        "living_antags": 1,
        "dead_players": 0,
        "observers": 0
      },
      
    ],
    "starting_threat_level": 77.6,
    "roundstart_rulesets": "Syndicate Traitors, ",
    "roundstart_pop": 1,
    "successful_injections": [
      
    ]
  }
}
```

**Review by gbasood — APPROVED** [review]

**Pull request merged**